packages/core/echo/echo-db/src/core-db/core-database.ts

```diff
@@ -49,6 +49,7 @@ export class CoreDatabase {
   readonly _updateEvent = new Event<ItemsUpdatedEvent>();
 
   private _isOpen = false;
```

Comment:
Is it critical that we differentiate openning, vs open states?

If so create an enum with CLOSED, OPENING, OPEN, CLOSING, variants and use it instead of the booleans